src/main/java/net/minecraftforge/items/pickup/ItemPickupReasons.java

```diff
@@ -5,21 +5,70 @@
 
 package net.minecraftforge.items.pickup;
 
-import static net.minecraftforge.items.pickup.ItemPickupReason.getOrCreate;
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.world.Container;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.vehicle.MinecartHopper;
+import net.minecraft.world.level.block.entity.HopperBlockEntity;
+
+import java.util.Collection;
+import java.util.Collections;
 
 /**
  * Class containing all standard ItemPickupReasons provided by Forge.
- *
- * @see ItemPickupReasons
+ * <p>This interface <b>should not</b> be custom implemented, instead make use of {@link #getOrCreate(String)}
  */
-public final class ItemPickupReasons
+@ApiStatus.NonExtendable
+public interface ItemPickupReasons
 {
-    /** Player is attempting to pick up an item */
-    public static final ItemPickupReason PLAYER = getOrCreate("player");
+    /**
+     * Player is attempting to pick up an item
+     * <p>{@code collector} types <i>should</i> result in the {@link Player} attempting to pick up the item.
+     */
+    ItemPickupReason PLAYER = getOrCreate("player");
+
+    /**
+     * LivingEntity is attempting to pick up an item
+     * <p>{@code collector} types <i>should</i> result in the {@link LivingEntity} attempting to pick up the item.
+     */
+    ItemPickupReason LIVING_ENTITY = getOrCreate("living_entity");
+
+    /**
+     * Hopper is attempting to suck in an item
+     * <p>{@code collector} types <i>should</i> result in the {@link HopperBlockEntity} or {@link MinecartHopper} {@link Container containers} attempting to pick up the item.
+     */
+    ItemPickupReason HOPPER = getOrCreate("hopper");
 
-    /** LivingEntity is attempting to pick up an item */
-    public static final ItemPickupReason LIVING_ENTITY = getOrCreate("living_entity");
+    /**
+     * Returns all registered pickup reasons.
+     *
+     * @return All registered pickup reasons.
```

Comment:
This should use the inline `{@return}` tag.

```suggestion
     * {@return all registered pickup reasons}
```